Sources/WalletConnectSign/Namespace.swift

```diff
@@ -47,6 +57,17 @@ enum Namespace {
             }
         }
     }
+    
+    static func validate(_ namespaces: [String: OptionalNamespace]) throws {
+        for (_, namespace) in namespaces {
+            if namespace.methods.isEmpty {
+                throw WalletConnectError.unsupportedNamespace(.unsupportedMethods)
+            }
+            if namespace.events.isEmpty {
+                throw WalletConnectError.unsupportedNamespace(.unsupportedEvents)
```

Comment:
is it always required to have an event in optional namespace?

Reply:
<img width="1033" alt="Screenshot 2023-01-30 at 11 38 28" src="https://user-images.githubusercontent.com/15164826/215454395-852f1355-0ea8-4522-816d-f7d31e4894c9.png">
In the doc the whole namespace structure is covered regardless of being `required` or `optional` so I think so.

Reply:
should the optional namespace contain the chains too then?
I don't think events and methods cannot be empty but let's discuss it today on the w3w meeting.